### ISSUE TITLE
floating point comparison

### ISSUE THREAD

**Issue by jwaterfield** [snoplusuk/echidna#79]
Change all floating point comparisons to use numpy.allclose instead of ==. The latter is causing many bugs. int cases can still use ==  / != e.g. number of bins

**Comment by ashleyrback** [comment]
Agreed. `==` has caused a lot of bugs. For completeness I'll mention the [Decimal](https://docs.python.org/2/library/decimal.html) module, which lets you perform both floating and fixed-point arithmetic on class instances. But I think converting all the relevant parameters to `Decimal` instances, may be more time consuming that it is worth in this case.
